<h3 id="exercise">Exercise 1 (Types): Your First Program 2023101086 이용관</h3>
숙제 제출시 반드시 "Exercise 1 (Types) : Your First Program"이라는 것과 함께, 자신의 이름과 학번을 꼭 기재하세요 (comment도 좋고, 아니면 파일이름에라도 좋고)

<p>Generations of programmers have started their coding careers by simply printing "Hello, world!". You will be following in their footsteps.</p>
<p>In the code cell below, use the <code>print()</code> function to print out the phrase: <code>Hello, world!</code></p>

In [34]:
print("Hello, world!")

Hello, world!


<p>Now, let's enhance your code with a comment. In the code cell below, print out the phrase: <code>Hello, world!</code> and comment it with the phrase <code>Print the traditional hello world</code> all in one line of code.</p>

In [36]:
print("Hello, world!") # Print the traditional hello world


Hello, world!


<hr>

<h3 id="exercise">Exercise 2 (Types): Get the object type</h3>

<p>In the code cell below, use the <code>type()</code> function to check the object type of <code>12.0</code>.

In [37]:
type(12.0)

float

In [39]:
# Convert a float to a string

str(1.2)

'1.2'

<h3 id="bool">Boolean data type</h3>

<p><i>Boolean</i> is another important type in Python. An object of type <i>Boolean</i> can take on one of two values: <code>True</code> or <code>False</code>:</p>

In [115]:
# Value true

True

True

<p>Notice that the value <code>True</code> has an uppercase "T". The same is true for <code>False</code> (i.e. you must use the uppercase "F").</p>

In [116]:
# Value false

False

False

<p>When you ask Python to display the type of a boolean object it will show <code>bool</code> which stands for <i>boolean</i>:</p> 

In [117]:
# Type of True

type(True)

bool

In [118]:
# Type of False

type(False)

bool

In [122]:
type(True)

bool

<p>We can cast boolean objects to other data types. If we cast a boolean with a value of <code>True</code> to an integer or float we will get a one. If we cast a boolean with a value of <code>False</code> to an integer or float we will get a zero. Similarly, if we cast a 1 to a Boolean, you get a <code>True</code>. And if we cast a 0 to a Boolean we will get a <code>False</code>. Let's give it a try:</p> 

In [123]:
# Convert True to int

int(True)

1

In [124]:
# Convert 1 to boolean

bool(1)

True

In [125]:
# Convert 0 to boolean

bool(0)

False

In [126]:
# Convert True to float

float(True)

1.0

<h3 id="exer_type">Exercise 3 (Types): Types</h3>

<p>What is the data type of the result of: <code>6 / 2</code>?</p>

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __here__ for the solution.

<!-- Your answer is below:
type(6/2) # float
-->

<p>What is the type of the result of: <code>6 // 2</code>? (Note the double slash <code>//</code>.)</p>

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __here__ for the solution.

<!-- Your answer is below:
type(6//2) # int, as the double slashes stand for integer division 
-->

<hr>

<h2 id="expressions">Expression and Variables</h2>

<h3 id="exp">Expressions</h3>

<p>Expressions in Python can include operations among compatible types (e.g., integers and floats). For example, basic arithmetic operations like adding multiple numbers:</p>

In [134]:
# Addition operation expression

43 + 60 + 16 + 41


160

<p>We can perform subtraction operations using the minus operator. In this case the result is a negative number:</p>

In [135]:
# Subtraction operation expression

50 - 60

-10

<p>We can do multiplication using an asterisk:</p>

In [136]:
# Multiplication operation expression

5 * 5

25

<p>We can also perform division with the forward slash:

In [137]:
# Division operation expression

25 / 5

5.0

In [138]:
# Division operation expression

25 / 6

4.166666666666667

<p>As seen in the quiz above, we can use the double slash for integer division, where the result is rounded to the nearest integer:

In [139]:
# Integer division operation expression

25 // 5

5

In [140]:
# Integer division operation expression

25 // 6

4

<h3 id="exer_exp">Exercise 4 (Types): Expression</h3>

<p>Let's write an expression that calculates how many hours there are in 160 minutes:

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __here__ for the solution.

<!-- Your answer is below:
160/60 
# Or 
160//60
-->

<p>Python follows well accepted mathematical conventions when evaluating mathematical expressions. In the following example, Python adds 30 to the result of the multiplication (i.e., 120).

In [141]:
# Mathematical expression

30 + 2 * 60

150

<p>And just like mathematics, expressions enclosed in parentheses have priority. So the following multiplies 32 by 60.

In [142]:
# Mathematical expression

(30 + 2) * 60

1920

<h3 id="var">Variables (변수)</h3>

<p>Just like with most programming languages, we can store values in <i>variables</i>, so we can use them later on. For example:</p>

In [148]:
# Store value into variable

x = 43 + 60 + 16 + 41

<p>To see the value of <code>x</code> in a Notebook, we can simply place it on the last line of a cell:</p>

In [149]:
# Print out the value in variable

x

160

<p>We can also perform operations on <code>x</code> and save the result to a new variable:</p>

In [151]:
# Use another variable to store the result of the operation between variable and value

y =  x / 60
y

2.6666666666666665

<p>If we save a value to an existing variable, the new value will overwrite the previous value:</p>

In [152]:
# Overwrite variable with new value

x = x / 60
x

2.6666666666666665

<p>It's a good practice to use meaningful variable names, so you and others can read the code and understand it more easily:</p>

In [153]:
# Name the variables meaningfully

total_min = 43 + 42 + 57 # Total length of albums in minutes
total_min

142

In [154]:
# Name the variables meaningfully

total_hours = total_min / 60 # Total length of albums in hours 
total_hours

2.3666666666666667

In [155]:
type(total_hours)

float

In [156]:
type(total_min)

int

<p>In the cells above we added the length of three albums in minutes and stored it in <code>total_min</code>. We then divided it by 60 to calculate total length <code>total_hours</code> in hours. You can also do it all at once in a single expression, as long as you use parenthesis to add the albums length before you divide, as shown below.</p>

In [157]:
# Complicate expression

total_hours = (43 + 42 + 60) / 60  # Total hours in a single expression
total_hours

2.4166666666666665

In [158]:
total_hours = 1000/3
total_hours

333.3333333333333

<p>If you'd rather have total hours as an integer, you can of course replace the floating point division with integer division (i.e., <code>//</code>).</p>

<h3 id="exer_exp_var">Exercise 5 (Types): Expression and Variables in Python</h3>

<p>What is the value of <code>x</code> where <code>x = 3 + 2 * 2</code></p>

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __here__ for the solution.

<!-- Your answer is below:
7
-->


<p>What is the value of <code>y</code> where <code>y = (3 + 2) * 2</code>?</p>

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __here__ for the solution.

<!-- Your answer is below:
10
-->

<p>What is the value of <code>z</code> where <code>z = x + y</code>?</p>

In [ ]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell

Double-click __here__ for the solution.

<!-- Your answer is below:
17
-->

<hr>
<h2>The last exercise!</h2>
<p>Congratulations, you have completed your first lesson and hands-on lab in Python. However, there is one more thing you need to do. The Data Science community encourages sharing work. The best way to share and showcase your work is to share it on GitHub. By sharing your notebook on GitHub you are not only building your reputation with fellow data scientists, but you can also show it off when applying for a job. Even though this was your first piece of work, it is never too early to start building good habits. Please submit all your exercises to Github & let the traces in our LMS, so for me to access your work. 
<hr>

<hr>
<p>Copyright &copy; 2023 RYU HOKYOUNG based on 2018 IBM Developer Skills Network. This notebook and its source code are released under the terms of the <a href="https://cognitiveclass.ai/mit-license/">MIT License</a>.</p>